packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -730,14 +753,23 @@ Future<DateTime> showDatePicker({
   @required BuildContext context,
   @required DateTime initialDate,
   @required DateTime firstDate,
-  @required DateTime lastDate
+  @required DateTime lastDate,
+  SelectableDayPredicate selectableDayPredicate
 }) async {
+  assert(!initialDate.isBefore(firstDate), 'initialDate must be on or after firstDate');
```

Comment:
ya, the tracking was a bit obscure in https://github.com/dart-lang/sdk/issues/6190 but it does work now